### ISSUE TITLE
Timeout error when using hostname in CORE_URL

### ISSUE THREAD

**Issue by mike0sv** [feast-dev/feast#1165]
## Expected Behavior 

Client works when CORE_URL (and other URLs) is provided either as ip:port or hostname:port

## Current Behavior

Only works with ip:port, if I use hostname:port, I get this stacktrace

```
~/anaconda3/lib/python3.7/site-packages/feast/grpc/grpc.py in create_grpc_channel(url, enable_ssl, enable_auth, ssl_server_cert_path, auth_metadata_plugin, timeout)
     50     except grpc.FutureTimeoutError:
     51         raise ConnectionError(
---> 52             f"Connection timed out while attempting to connect to {url}"
     53         )
```

## Steps to reproduce

Provide CORE_URL as hostname:port
Maybe there is something special about my network (it is on-premise network and I dont have any details about it), but `curl hostname:port` works (it gives an error obviously, but the connection is established)

### Specifications

- Version: 0.8.0
- Platform: linux
- Subsystem: python sdk

**Comment by woop** [comment]
@mike0sv Are you curling the same port as Feast uses?

**Comment by mike0sv** [comment]
Yes, it's default 6565. I get `curl: (1) Received HTTP/0.9 when not allowed`

**Comment by woop** [comment]
To eliminate a DNS issue, what happens when you hardcode the DNS entry in `/etc/hosts`?

**Comment by mike0sv** [comment]
If i put it there, it works. Also, I noticed that when I `ping hostname`, it resolves as `hostname.xxx.local`. If I use this as core_url, everything also works.
I also checked and both requests and urllib modules are able to resolve just `hostname`, without `.xxx.local`. So maybe it's grpc issue, but I don't know how to check it fast.
Btw, I'm doing all of this from a docker container, and feast services are on a different machine, but also in docker (ports exposed obviously), just FYI
